server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutes.java

```diff
@@ -173,14 +181,48 @@ public void define(Service service) {
         @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500, message = "Internal server error - Something went bad on the server side.")
     })
     private TaskIdDto userActions(Request request, Response response) throws JsonExtractException {
-        UserVaultAction requestedAction = extractUserVaultAction(request);
+        VaultAction requestedAction = extractVaultAction(request);
 
         Task requestedTask = generateTask(requestedAction, request);
         TaskId taskId = taskManager.submit(requestedTask);
         return TaskIdDto.respond(response, taskId);
     }
 
-    private Task generateTask(UserVaultAction requestedAction, Request request) throws JsonExtractException {
+    @POST
+    @Path(ROOT_PATH)
+    @ApiOperation(value = "Purge all expired messages based on retentionPeriod of deletedMessageVault configuration")
+    @ApiImplicitParams({
+        @ApiImplicitParam(
+            required = true,
+            name = "action",
+            dataType = "String",
+            paramType = "query",
+            example = "?action=purge",
+            value = "Compulsory. Needs to be a purge action")
+    })
+    @ApiResponses(value = {
+        @ApiResponse(code = HttpStatus.CREATED_201, message = "Task is created", response = TaskIdDto.class),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Bad request - action is invalid"),
+        @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500, message = "Internal server error - Something went bad on the server side.")
+    })
+    private TaskIdDto globalActions(Request request, Response response) {
+        VaultAction vaultAction = extractVaultAction(request);
+
+        TaskId taskId = taskManager.submit(generateVaultTask(vaultAction));
+        return TaskIdDto.respond(response, taskId);
+    }
+
+    private Task generateVaultTask(VaultAction requestedAction) {
+        switch (requestedAction) {
+            case PURGE:
+                validatePurgeAction(requestedAction);
```

Comment:
This is not really useful as if we get here, we know for sure we have a `PURGE`

I propose

```
private Task generateVaultTask(VaultAction requestedAction) {
    validatePurgeAction(requestedAction);
    return deletedMessageVault.deleteExpiredMessagesTask();
}
```

Simply....